Add docs for dropdowns (#3)
DV8FromTheWorld/JDA-Website: docs/using-jda/interactions.md

    ```

# Component Interactions

To add components to a message you can use up to 5 ActionRows.

You can add multiple ActionRows with either `setActionRows` or `addActionRows` (depending on whether you are sending or editing a message).
For the common case of a single ActionRow you can also use `setActionRow(Component...)` or `addActionRow(Component...)`.

Each ActionRow can hold up to a certain amount of components:
- 5 Buttons
- 1 Selection Menu (Dropdown)

These component interactions offer 4 response types:

- Reply
- Deferred Reply
- Edit Message
- Deferred Edit Message

> The reply and deferred reply responses are identical to the Slash-Commands response types. However, these new edit response types are used to update the existing message the component is attached to. If you simply want to acknowledge that the component was successfully interacted with, you can simply do a `deferEdit()` without any further updates, which will prevent the interaction from failing on the user side.

Reviewer: ```suggestion
The reply and deferred reply responses are identical to the Slash-Commands response types. However, these new edit response types are used to update the existing message the component is attached to. If you just want to acknowledge that the component was successfully interacted with, you can simply call `deferEdit()` without any further updates, which will prevent the interaction from failing on the user side.
```

To properly use an interactive component, you need to use the **Component ID** (aka **Custom ID**).
This ID can also be used to then *identify* which component was pressed by the user.

> Such Component ID is provided by `getComponentId()` on every Component Interaction.

Reviewer: It might be worth mentioning these must be unique across all components in a message

                 Button.primary("hello", "Click Me"), // Button with only a label
                 Button.success("emoji", Emoji.fromMarkdown("<:minn:245267426227388416>"))) // Button with only an emoji
               .queue();
       } else if (event.getName().equals("info")) {
           event.reply("Click the buttons for more info")
               .addActionRow( // link buttons don't send events, they just open a link in the browser when clicked
                   Button.link("https://github.com/DV8FromTheWorld/JDA", "GitHub")
                     .withEmoji(Emoji.fromMarkdown("<:github:849286315580719104>")), // Link Button with label and emoji
                   Button.link("https://ci.dv8tion.net/job/JDA/javadoc/", "Javadocs")) // Link Button with only a label
               .queue();
       }
   }

   @Override
   public void onButtonClick(ButtonClickEvent event) {
       if (event.getComponentId().equals("hello")) {
           event.reply("Hello :)").queue(); // send a message in the channel
       } else if (event.getComponentId().equals("emoji")) {
           event.editMessage("That button didn't say click me").queue(); // update the message
       }
   }
 }

> ```

Reviewer 2: This section has to be indented by 4 spaces, like it originally was, for it to show as part of the admonition.

## Selection Menus (Dropdowns)

Every dropdown can be disabled and have up to 25 options.

It's possible to set the minimum and maximum amount of options to be selected.

Each option can have its own label, description, and emoji.
There can be multiple options selected and set as default.

![Example Selection Menu With A Default Value](https://i.imgur.com/44q006n.png)

> ### Handling SelectionMenuEvent
> 
> When a user chooses options from a dropdown, you will receive a `SelectionMenuEvent` for the respective interaction with the selected values.

Reviewer: ```suggestion
### Handling SelectionMenuEvent

When a user selects their options from a dropdown and submits their choices, you will receive a `SelectionMenuEvent` for the respective interaction with the selected values.
```
I think it should be clear that users just clicking on options will not fire the events

!!! example

```java
 public class DropdownBot extends ListenerAdapter {
	@Override
	public void onSlashCommand(SlashCommandEvent event) {
		if (event.getName().equals("food")) {
			event.reply("Choose your favorite food")
				.addActionRow(
					SelectionMenu.create("choose-food")
					  .addOption("Pizza", "pizza", "Classic") // SelectOption with only the label, value, and description
					  .addOptions(SelectOption.of("Hamburger", "hamburger") // another way to create a SelectOption
							.withDescription("Tasty") // this time with a description
							.withEmoji(Emoji.fromUnicode("\uD83C\uDF54")) // and an emoji
							.withDefault(true)) // while also being the default option
					.build())
				.queue();
		}
	}
  
	@Override
	public void onSelectionMenu(SelectionMenuEvent event) {
		if (event.getComponentId().equals("choose-food")) {
			event.reply("You chose " + event.getValues().get(0)).queue();
		}
	}
 }

> ```

Reviewer 2: This code block also needs 4 spaces of indentation to fit inside the admonition

Reviewer 3: This is indented with Tabs.
Please convert the tabs to spaces. We should not have tabs in the project. If there are any tabs throughout the project we should convert them to spaces @Aikovdp

Reviewer 2: I've checked, and it looks like there aren't any tabs anywhere, they were only added in this PR. Once Cane replaces his tabs with spaces we'll be all good.
